app/src/services/read-only.service.js

```diff
@@ -0,0 +1,29 @@
+module.exports = class ReadOnlyService {
+
+    constructor(blacklist = [], whitelist = []) {
+        this.blacklist = blacklist;
+        this.whitelist = whitelist;
+    }
+
+    isBlacklisted(path) {
+        return this.blacklist.some((element) => new RegExp(element).test(path));
+    }
+
+    isWhitelisted(path) {
+        return this.whitelist.some((element) => new RegExp(element).test(path));
+    }
+
+    shouldBlockRequest(method, path) {
+        return (ReadOnlyService.isRead(method) && this.isBlacklisted(path))
+        || (ReadOnlyService.isWrite(method) && !this.isWhitelisted(path));
+    }
+
+    static isRead(method) {
+        return method === 'GET';
+    }
+
+    static isWrite(method) {
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods

Reply:
Leaving this as is, because we only support these methods as write methods (supported methods are defined in the CORS plugin).